src/app/lib/business/logger/logger_manager.dart

```diff
@@ -0,0 +1,221 @@
+  late bool _initialIsFileLoggingEnabled;
+
+  @override
+  late bool isConsoleLoggingEnabled;
+
+  @override
+  late bool isFileLoggingEnabled;
+
+  @override
+  bool hasConfigurationBeenChanged = false;
+
+  _LoggerManager(this._loggerRepository);
+
+  @override
+  Future<Logger> createLogInstance() async {
+    final List<LogOutput> loggerOutputs = [];
+
+    final loggingConfiguration =
+        await _loggerRepository.getLoggingConfiguration();
+
+    _initialIsConsoleLoggingEnabled =
+        loggingConfiguration.isConsoleLoggingEnabled ??
+            bool.parse(dotenv.env["IS_CONSOLE_LOGGING_ENABLED"] ?? 'false');
+    if (_initialIsConsoleLoggingEnabled) {
+      loggerOutputs.add(CustomConsoleOutput());
+    }
+    isConsoleLoggingEnabled = _initialIsConsoleLoggingEnabled;
+
+    _initialIsFileLoggingEnabled = loggingConfiguration.isFileLoggingEnabled ??
+        bool.parse(dotenv.env["IS_FILE_LOGGING_ENABLED"] ?? 'false');
+    if (_initialIsFileLoggingEnabled) {
+      // Initialize the log file.
+      final Directory appDocumentsDir =
+          await getApplicationDocumentsDirectory();
+      _logFile = File('${appDocumentsDir.path}/$_fileName');
+      loggerOutputs.add(CustomFileOutput(file: _logFile!));
+    }
+    isFileLoggingEnabled = _initialIsFileLoggingEnabled;
+
+    var minimumLevel = dotenv.env['MINIMUM_LEVEL']!.toLogLevel();
+    return _logger = Logger(
+      filter: LevelLogFilter(minimumLevel),
+      printer: PrefixPrinter(
+        PrettyPrinter(
+          methodCount: 0,
+          errorMethodCount: 9,
+          stackTraceBeginIndex: 1,
+          printTime: true,
+          printEmojis: false,
+          noBoxingByDefault: true,
+        ),
+      ),
+      output: MultiOutput(loggerOutputs),
+    );
+  }
+
+  @override
+  Future<bool> deleteLogFile() async {
+    _logger.d("Start deleting the log file.");
+    if (_logFile != null && await _logFile!.exists()) {
+      try {
+        _logger.t("Log file exists. It's time to delete it.");
+
+        await _logFile!.delete();
+
+        _logger.i("The log file was deleted successfully.");
+        return true;
+      } catch (e, stackTrace) {
+        _logger.e(
+          'Failed to delete the log file.',
+          error: e,
+          stackTrace: stackTrace,
+        );
+      }
+    }
+
+    _logger.w('No log file to delete.');
+    return false;
+  }
+
+  @override
+  Future<bool> shareLogFile() async {
+    _logger.d("Start sharing the log file.");
+    if (_logFile != null &&
+        (await _logFile!.exists()) &&
+        (await _logFile!.length()) > 0) {
+      try {
+        _logger.t("Log file exists. It's time to share it.");
+
+        var result = await Share.shareXFiles(
+          [XFile(_logFile!.path)],
+          subject: "Diagnostics - ApplicationTemplate ${DateTime.now()}",
+        );
+
+        _logger.i(
+            "Share execution has been successful. It returned the following status ${result.status}");
```

Comment:
nitpick but I wonder if the end of the second sentence should have ponctuation

Reply:
It definitly should. I forgot to put one :S